Include core-js package in package.json

Demivan opened the pull request:
Fixes #332

Grsmto commented:
Hi,
Thanks for your contribution.
Does this actually fixes the issue? As I suppose you'll have both core-js 2 and 3 as dependency but they can't work concurrently so I'm not sure to see how that would fix it?
If you check the `simplebar` package, there is the corejs 3 configuration in place there. Maybe there should just be the same thing in the vue package?

Demivan commented:
Yeah. It fixed this issue for me.
Right now issue is that `simplebar-vue` does not have `core-js` package specified as dependency at all. It just uses hoisted package version, so if app has `core-js@3` in its `package.json` it will try to use that. It does not use `core-js@3` dependency from `simplebar` package. And even if it did it would not work because current babel config for `simplebar-vue` generates `core-js` imports using version 2 format.
Best solution would, probably, be to update vue related dependencies (babel preset) for `simplebar-vue` and switch to `core-js@3`.
I can update this pull-request to include that.

Demivan commented:
Just checked vue packages and `babel-preset-app@3.8.0` is not updated to `core@3` yet, version 4 where it is updated is in alpha.

Demivan commented:
I have removed `@vue/cli-service` and related packages - they were used mostly for tests. Instead I have added `vue-jest` and used `jest` directly for tests. Additionally I have removed `@vue/babel-preset-app` and used just `@babel/preset-env`. This allowed me to switch to `core-js@3`.
Tests are passing, but I will test tomorrow on my project for both `core-js@2` and `core-js@3`
Another benefit of these changes are 3.5k lines removed from yarn.lock :slightly_smiling_face:

Demivan commented:
Just tested and it worked in both cases: when app is targeting core-js@2 and when core-js@3

Grsmto commented on packages/simplebar-vue/__snapshots__/index.test.js.snap:
```diff
@@ -1,49 +1,110 @@
+<div
+  data-simplebar=""
+  data-simplebar-auto-hide="false"
+>
```
hey, do you know why these changes happened? It seems like a prettifier passed here.
In theory we have a precommit hook with Prettier so idk what happened. if you could revert this before merge.

Demivan replied:
It was because I used `wrapper.element` when making snapshots.
https://vue-test-utils.vuejs.org/guides/testing-single-file-components-with-jest.html#snapshot-testing
I added `jest-serializer-vue` now so snapshots are the same as before.

Grsmto replied:
Hey actually I thought this was the source, I didn't realise I was commenting on a snapshot file!
That's totally fine to have these changes then haha
